tests/qunit/ModelTest.js

```diff
@@ -0,0 +1,26 @@
+require( '../../assets/js/Model' );
+
+QUnit.test( 'source and target languages', function ( assert ) {
+	var model = new App.Model();
+	model.setSourceLang( 'fr' );
+	assert.strictEqual( 'fr', model.getSourceLang() );
+} );
+
+QUnit.test( 'get source translation labels', function ( assert ) {
+	var model = new App.Model( {
+		n1: {
+			fr: { text: 'Foo' },
+			fallback: { text: 'Bar' }
+		}
+	} );
+	model.setSourceLang( 'fr' );
+	assert.deepEqual(
+		model.getSourceTranslation( 'n1' ),
+		{ label: 'Foo', class: '' }
+	);
+	model.setSourceLang( 'eo' );
+	assert.deepEqual(
+		model.getSourceTranslation( 'n1' ),
+		{ label: 'Bar', class: 'source-lang-not-found' }
+	);
+} );
```

Comment:
You can add event-driven tests to make sure the event is emitted as planned (and not emitted redundantly) I think we did this before for TemplateWizard? If not, the general pattern is:
- Create an array
- Listen to the required event, and in the event listener, push data into the array (like the name of the language)
- Do a bunch of language changes, including ones that shouldn't emit events (like double-changing the same language?)
- Check that the content of the array is what we expect

Example here: https://github.com/wikimedia/mediawiki/blob/master/tests/qunit/suites/resources/mediawiki.rcfilters/dm.FilterItem.test.js#L156

This will become more crucial when we have more events on the model, so you can leave this for the followup if it becomes too cumbersome for this one, but when we have several events (and several widgets that set/listen to events) then it becomes useful to check that events are emitted as expected.